codegen/templates/tchannel_endpoint.tmpl

```diff
@@ -0,0 +1,250 @@
+{{- /* template to render edge gateway tchannel server code */ -}}
+// Code generated by zanzibar
+// @generated
+
+// Package {{.PackageName}} is generated code used to handle TChannel calls using Thrift.
+package {{.PackageName}}
+
+import (
+	"context"
+	"errors"
+
+	"go.uber.org/thriftrw/wire"
+	"go.uber.org/zap"
+	"{{.GatewayPackageName}}/clients"
+	zanzibar "github.com/uber/zanzibar/runtime"
+
+	{{range $idx, $pkg := .IncludedPackages -}}
+	{{$pkg.AliasName}} "{{$pkg.PackageName}}"
+	{{end -}}
+
+	{{if .Method.Downstream -}}
+	{{- range $idx, $pkg := .Method.Downstream.IncludedPackages -}}
+	{{$file := basePath $pkg.PackageName -}}
+	{{$pkg.AliasName}} "{{$pkg.PackageName}}"
+	{{end}}
+	{{- end}}
+)
+
+{{$workflow := printf "TChan%s" .WorkflowName -}}
+{{$reqHeaderMap := .ReqHeaderMap -}}
+{{$reqHeaderMapKeys := .ReqHeaderMapKeys -}}
+{{$resHeaderMap := .ResHeaderMap -}}
+{{$resHeaderMapKeys := .ResHeaderMapKeys -}}
+{{$clientName := title .ClientName -}}
+{{$serviceMethod := printf "%s%s" .Method.ThriftService .Method.Name -}}
+{{$handlerName := printf "%sHandler"  $serviceMethod -}}
+{{$genCodePkg := .Method.GenCodePkgName -}}
+{{with .Method -}}
+// New{{$handlerName}} creates a handler to be registered with a thrift server.
+func New{{$handlerName}}(
+	gateway *zanzibar.Gateway,
+) zanzibar.TChanHandler {
+	return &{{$handlerName}}{
+		Clients: gateway.Clients.(*clients.Clients),
+		Logger: gateway.Logger,
+	}
+}
+
+// {{$handlerName}} is the handler for "{{.ThriftService}}::{{.Name}}".
+type {{$handlerName}} struct {
+	Clients *clients.Clients
+	Logger *zap.Logger
+}
+
+// Service returns the service name.
+func (h *{{$handlerName}}) Service() string {
+	return "{{.ThriftService}}"
+}
+
+// Method returns the method name.
+func (h *{{$handlerName}}) Method() string {
+	return "{{.Name}}"
+}
+
+// Handle handles RPC call of "{{.ThriftService}}::{{.Name}}".
+func (h *{{$handlerName}}) Handle(
+	ctx context.Context,
+	reqHeaders map[string]string,
+	wireValue *wire.Value,
+) (bool, zanzibar.RWTStruct, map[string]string, error) {
+	var res {{$genCodePkg}}.{{.ThriftService}}_{{.Name}}_Result
+
+	{{if ne .RequestType "" -}}
+	var req {{.RequestType}}
+	if err := req.FromWire(*wireValue); err != nil {
+		return false, nil, nil, err
+	}
+	{{end -}}
+
+	workflow := {{$workflow}}{
+		Clients: h.Clients,
+		Logger: h.Logger,
+	}
+
+	{{if and (eq .RequestType "") (eq .ResponseType "")}}
+	respHeaders, err := workflow.Handle(ctx, headers)
+	{{else if eq .RequestType ""}}
+	r, respHeaders, err := workflow.Handle(ctx, reqHeaders)
+	{{else if eq .ResponseType ""}}
+	respHeaders, err := workflow.Handle(ctx, reqHeaders, &req)
```

Comment:
`reqHeaders` here should implement `zanzibar.Headers` interface instead of `map[string]string`